<!-- https://github.com/ros-controls/ros_controllers/tree/kinetic-devel/diff_drive_controller/test -->
<launch>
    <!--arg name="model" default="$(env sambot_MODEL)" doc="model type [sambot, remo]"/-->
    <arg name="map_file" default="$(find sambot_base)/maps/map.yaml"/>
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)"/>
    
   <param name="/sambot1/robot_description" command="$(find xacro)/xacro --inorder '$(find sambot_description)/urdf/sambot_xacro.xacro'"/>
   <param name="/sambot2/robot_description" command="$(find xacro)/xacro --inorder '$(find sambot_description)/urdf/sambot_xacro.xacro'"/>
   <param name="/sambot3/robot_description" command="$(find xacro)/xacro --inorder '$(find sambot_description)/urdf/sambot_xacro.xacro'"/>
   <param name="/sambot4/robot_description" command="$(find xacro)/xacro --inorder '$(find sambot_description)/urdf/sambot_xacro.xacro'"/>

   <!-- <group ns="sambot1">
        <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
            <param name="output_frame" value="sambot1/odom"/>
            <param name="base_footprint_frame" value="sambot1/chassis"/>
            <param name="freq" value="30.0"/>
            <param name="sensor_timeout" value="1.0"/>  
            <param name="odom_used" value="true"/>
            <param name="imu_used" value="true"/>
            <param name="vo_used" value="false"/>
            <remap from="odom" to="mobile_base_controller/odom"/>
            <remap from="imu_data" to="imu/data" />    
        </node>
   </group> -->


   <!-- <node pkg="sambot_base" type="odom_transform_publisher" name="odom_transform" /> -->
   <!-- <node pkg="sambot_base" type="cmd_vel_broadcaster" name="cmd_vel_broadcaster" /> -->

   <arg name="in_update_trigger_topic" default="/trigger_field_update"/>
    <arg name="out_field_topic" default="/spatial_temporal_field"/>
    <arg name="out_agent_field_topic" default="/agent_field"/>
    <arg name="initial_field_state_file" default="$(find sambot_rl)/data/initial_field/u_npy.npy"/>
    <arg name="dx" default="-2.5"/>
    <arg name="dy" default="-2.5"/>
    <arg name="vx" default="-0.6"/>
    <arg name="vy" default="0.8"/>
    <arg name="dt" default="0.1"/>
    <arg name="k" default="1.0"/>
    <arg name="num_rows" default="100"/>
    <arg name="num_cols" default="100"/>
    <arg name="grid_cell_size" default="0.05"/>

    <!-- <node pkg="tf" type="static_transform_publisher" name="map_odom_broadcaster" 
    args="0 0 0 0 0 0 map sambot1/odom 100"/> -->

    <!-- <node pkg="sambot_base" type="MoveToGoalActionServer" name="MoveToGoalActionServer" ns="sambot1" /> -->
    <node name="camera_publisher_node" pkg="sambot_rl" type="camera_publisher.py" output="screen"></node>


    <node pkg="sambot_rl" name="spatial_temporal_field_publisher_node"
        type="spatial_temporal_field_publisher_node" output="screen">

        <param name="in_update_trigger_topic" value="$(arg in_update_trigger_topic)"/>
        <param name="out_field_topic" value="$(arg out_field_topic)"/>
        <param name="initial_field_state_file" value="$(arg initial_field_state_file)"/>
        <param name="out_agent_field_topic" value="$(arg out_agent_field_topic)"/>
        <param name="dx" value="$(arg dx)"/>
        <param name="dy" value="$(arg dy)"/>
        <param name="vx" value="$(arg vx)"/>
        <param name="vy" value="$(arg vy)"/>
        <param name="dt" value="$(arg dt)"/>
        <param name="k" value="$(arg k)"/>
        <param name="num_rows" value="$(arg num_rows)"/>
        <param name="num_cols" value="$(arg num_cols)"/>
        <param name="grid_cell_size" value="$(arg grid_cell_size)"/>

    </node>
</launch>